lib/ddtrace/tracer.rb

```diff
@@ -158,7 +155,8 @@ def default_service
     #
     #   tracer.set_tags('env' => 'prod', 'component' => 'core')
     def set_tags(tags)
-      @tags.update(tags)
+      string_tags = Hash[tags.collect { |k, v| [k.to_s, v] }]
```

Comment:
This fixes one of the bugs where Symbols and Strings weren't being quantized to the same key, causing duplicates.